python/kserve/kserve/protocol/infer_type.py

```diff
@@ -173,7 +211,14 @@ def as_numpy(self) -> np.ndarray:
         if dtype is None:
             raise InvalidInput(f"invalid datatype {dtype} in the input")
         if self._raw_data is not None:
-            np_array = np.frombuffer(self._raw_data, dtype=dtype)
+            if self.datatype == "BYTES":
+                # String results contain a 4-byte string length
+                # followed by the actual string characters. Hence,
+                # need to decode the raw bytes to convert into
+                # array elements.
+                np_array = deserialize_bytes_tensor(self._raw_data)
```

Comment:
How `deserialize_bytes_tensor` is different from `np.frombuffer` ?

Reply:
Usually strings are serialized as a 1-dimensional array containing the 4-byte byte size followed by the actual element bytes. All elements are concatenated together. To deserialize this format we are using deserialize_byte_tensor

Reply:
what happens if user set the string dtype np.array bytes to the raw content field?

Reply:
serialize_bytes_tensor only used if the strings is sent in raw_input_contents. If we use contents we can just directly provide strings as list of bytes (e.g. [b"cat", b"dog"] )